core/chains/evm/client/send_only_node.go

```diff
@@ -94,16 +115,24 @@ func (s sendOnlyNode) String() string {
 	return fmt.Sprintf("(secondary)%s:%s", s.name, s.uri.String())
 }
 
-func (s sendOnlyNode) Verify(ctx context.Context, expectedChainID *big.Int) (err error) {
-	if chainID, err := s.ChainID(ctx); err != nil {
+func (s sendOnlyNode) verify(parentCtx context.Context) (err error) {
+	s.log.Debugw("evmclient.Client#ChainID(...)")
+	ctx, cancel := s.wrapCtx(parentCtx)
+	defer cancel()
+	if chainID, err := s.geth.ChainID(ctx); err != nil {
 		return errors.Wrap(err, "failed to verify chain ID")
-	} else if chainID.Cmp(expectedChainID) != 0 {
+	} else if chainID.Cmp(s.chainID) != 0 {
 		return errors.Errorf(
 			"sendonly rpc ChainID doesn't match local chain ID: RPC ID=%s, local ID=%s, node name=%s",
 			chainID.String(),
-			expectedChainID.String(),
+			s.chainID.String(),
 			s.name,
 		)
 	}
 	return nil
 }
+
+func (s sendOnlyNode) wrapCtx(parentCtx context.Context) (combinedCtx context.Context, cancel context.CancelFunc) {
+	combinedCtx, cancel = utils.CombinedContext(parentCtx, s.ctx, queryTimeout)
+	return
```

Comment:
Do the return param names add value here? Or would this read easier?
```suggestion
func (s sendOnlyNode) wrapCtx(parentCtx context.Context) (context.Context, context.CancelFunc) {
	return utils.CombinedContext(parentCtx, s.ctx, queryTimeout)
```